### DISCUSSION TITLE
Do Metro / Gradle absolutely need to patch the Android SDK versions ?  (react-native & nixos) )

### DISCUSSION THREAD

**Discussion by Twiggeh** [react-native-community/discussions-and-proposals#609] [Ideas]
Hi! 

I am a huge fan of deterministic builds so I am using NixOS for my entire setup, unfortunately I cannot start any react-native projects due to it trying to patch build tools from 33.0.0 to 30.0.0 and vice verse ad infinitum & some form of pathcer;v4 SDK Patch Applier v4

Due to the deterministic nature, store files (where the android-sdk is currently located at), are immutable, hence why react-native fails. 

Is there a way to disable the patch / is the patching absolutely necessary ?

**Comment by cortinico** [comment]
> I cannot start any react-native projects due to it trying to patch build tools from 33.0.0 to 30.0.0 and vice verse ad infinitum & some form of pathcer;v4 SDK Patch Applier v4

Why is NixOS trying to patch build tools to 30.0.0? This seems like a NixOS issue and not a React Native one to me
